Lib/ufo2ft/fastUFO.py

```diff
@@ -228,8 +322,184 @@ def save(self, path=None, formatVersion=3):
+        if font is not None:
+            self._font = weakref.ref(font)
+        else:
+            self._font = None
+
+    @property
+    def font(self):
+        if self._font is not None:
+            return self._font()
+
+    @property
+    def defaultLayer(self):
+        return self._defaultLayer
+
+    @defaultLayer.setter
+    def defaultLayer(self, layer):
+        self._defaultLayer = layer
+        self._defaultLayerName = layer.name
+
+    @property
+    def layerOrder(self):
+        return list(self._layers.keys())
+
+    @layerOrder.setter
+    def layerOrder(self, order):
+        assert set(order) == set(self._layers)
+        new_layers = OrderedDict()
+        for layer_name in order:
+            new_layers[layer_name] = self._layers[layer_name]
+        self._layers = new_layers
+
+    def newLayer(self, name):
+        layer = Layer(name, self)
+        self._layers[name] = layer
+        return layer
+
+    def __iter__(self):
+        for name in self.layerOrder:
+            yield self[name]
+
+    def __getitem__(self, name):
+        if name is None:
+            name = self._defaultLayerName
+        return self._layers[name]
+
+    def __delitem__(self, name):
+        if name is None:
+            name = self._defaultLayerName
+        if name not in self:
+            raise KeyError('%s not in layers' % (name))
+        del self._layers[name]
+
+    def __len__(self):
+        return len(self._layers)
+
+    def __contains__(self, name):
+        if name is None:
+            name = self._defaultLayerName
+        return name in self._layers
+
+    def save(self, writer):
+        for layerName in self._layers:
+            defaultLayer = (layerName == self._defaultLayerName)
+            glyphset = writer.getGlyphSet(layerName, defaultLayer)
+            for name, glyph in sorted(self._layers[layerName].glyphs.items()):
+                glyphset.writeGlyph(name, glyph, glyph.drawPoints)
+            glyphset.writeContents()
         writer.writeLayerContents()
+
+
+class Layer(object):
+    def __init__(self, name, layerSet):
+        self.name = name
+        if layerSet is not None:
+            self._layerSet = weakref.ref(layerSet)
+        else:
+            self._layerSet = None
+        self.glyphs = {}
+
+    @property
+    def layerSet(self):
+        if self._layerSet is not None:
+            return self._layerSet()
+
+    def __contains__(self, name):
+        return name in self.glyphs
+
+    def __getitem__(self, name):
+        return self.glyphs[name]
+
+    def __iter__(self):
+        for name in self.keys():
+            yield self.glyphs[name]
+
+    def __delitem__(self, name):
+        del self.glyphs[name]
+
+    def keys(self):
+        return self.glyphs.keys()
+
+    def newGlyph(self, name):
+        glyph = Glyph(name, self)
+        glyph.layer = self
+        glyph.font = self.layerSet.font
+        self.glyphs[name] = glyph
+        return glyph
+
+
+class Anchor(Mapping):
```

Comment:
could we use regular dict instead? Or is some other code relying on such an object having both dict-like subscript and attribute lookup behavior?